### ISSUE TITLE
Prefer array forEach over _ in backend code

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/compiler-explorer#3003]
Tiny nit: ss there any reason for using _.each for an array on the backend over Array.forEach?

_Originally posted by @supergrecko in https://github.com/compiler-explorer/compiler-explorer/pull/2963#discussion_r716266640_

**Comment by mattgodbolt** [comment]
Specifically in the PR mentioend above

**Comment by Bhavin789** [comment]
Hi @mattgodbolt , looks like a simple replacement. Can I start working on this issue if that is ok? Thanks

**Comment by mattgodbolt** [comment]
Sure thing @Bhavin789 ! Thanks!

**Comment by Bhavin789** [comment]
Hi @mattgodbolt, I was going through the instances of `_.each` in the code base and could see that there are around 40 files that would be affected.

The issue which I found with the replacement is that, `_.each` can be used for objects as well and not just array. Since the repo is in JS, it is a bit difficult for me to see which instances are iterating over objects and which one over the array. Plus, replacing it all in one PR may lead to regressions.

ATM, my time won't permit me to have a look at this much scope. Apologies but I won't be able to work on this right now. Maybe I can have a look at it sometime later. Hope that is ok. Please feel free to re-assign it to someone else if someone willing to pick up.

Thanks!

**Comment by mattgodbolt** [comment]
Hi @Bhavin789 ! This change (at least initially) was meant to mean only the files that @supergrecko indicated in the linked issue report (`lib/options-handler.js`). So if you have only a short time available, we'd appreciate just the fixes there, if you have them.

More generally we only have `Array.forEach` available in the backend code, and in the typescript code coming.

Thanks for taking a look, I'll unassign you but any work you've done so far may still be useful to PR :)

**Comment by Bhavin789** [comment]
Thanks for the clarity @mattgodbolt. I badly misunderstood it. I thought it was meant for the whole codebase. I agree, we can start replacing one by one and I can initially start with the usage in the above-mentioned PR. 
Gradually, I can have a look at other files as well.

If this is ok, please feel free to re-assign it to me. Sorry for the misunderstanding. Thanks!

**Comment by Bhavin789** [comment]
Hi @mattgodbolt while replacement, I found that using `forEach` on array is actually violating a [lint rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v32.0.1/docs/rules/no-array-for-each.md).

![image](https://user-images.githubusercontent.com/15359575/136239626-22256563-94fd-4ac9-8220-ea4197f37efd.png)

Do I need to disable this rule or use `for of` ? Thanks!

**Comment by partouf** [comment]
> Sure @apmorton . Thanks for the clarity!
> 
> Meanwhile, I was trying to commit but some tests are failing for me even on the `main` branch with no changes. This is actually not allowing me to commit. Any idea on why this might be happening? Thanks
> 
> ![image](https://user-images.githubusercontent.com/15359575/136250203-c58d3450-90e3-48c2-a47d-3f692dda5841.png)

It looks like we're doing some c++filt unit tests on *nix platforms, but your platform (mac?) doesn't produce the same demangling.
I suggest you use `git commit --no-verify` to circumvent the unit tests temporarily while we figure this out.

**Comment by Bhavin789** [comment]
Yes @partouf, I am on a mac system. Will use the above command to raise a PR at least. Thanks for the quick help 😃
